docarray/array/array/io.py

```diff
@@ -311,21 +310,22 @@ def to_base64(
     @classmethod
     def from_json(
         cls: Type[T],
-        file: Union[str, bytes, bytearray],
+        file: str,
     ) -> T:
         """Deserialize JSON strings or bytes into a DocArray.
 
         :param file: JSON object from where to deserialize a DocArray
         :return: the deserialized DocArray
         """
         json_docs = json.loads(file)
-        return cls([cls.document_type.parse_raw(v) for v in json_docs])
+        return cls([cls.document_type(**v) for v in json_docs])
 
     def to_json(self) -> str:
         """Convert the object into a JSON string. Can be loaded via :meth:`.from_json`.
         :return: JSON serialization of DocArray
         """
-        return json.dumps([doc.json() for doc in self])
+        doc_jsons = ', '.join([doc.json() for doc in self])
```

Comment:
I think here we could just use orson dump on `self._data`

Reply:
I don't see how. Then docs are treated as strings (double-encoded)
Would be nice if you could specify custom encoders for specific types, like 
```python
orjson.dumps(docs, custom_encoders={BaseDoc: lambda x: x.json()}
```
but i don't think that's possible

Reply:
okay I see. I think this is feasible with the default thing

Reply:
Ok so this also works but not sure if it's better because we're not using pydantic's json() method anymore.

```python

class CustomEncoder(json.JSONEncoder):
    def default(self, val):
        if isinstance(val, BaseDoc):
            return val.dict()

def to_json(self) -> str:
    return json.dumps([doc for doc in self], cls=CustomEncoder)
```

wdyt?

Reply:
This way is better performance wise. We delegate everything to orjson. Otherwsie the `b''.join` need to copy paste everything into the new bytes. And this is expensive

Reply:
We're not doing bytes here.

Basically it's between these two:

```python
def to_json(self) -> str:
    doc_jsons = ', '.join([doc.json() for doc in self])
    return f'[{doc_jsons}]'
```
```python
class CustomEncoder(json.JSONEncoder):
    def default(self, val):
        if isinstance(val, BaseDoc):
            return val.dict()

def to_json(self) -> str:
    return json.dumps([doc for doc in self], cls=CustomEncoder)
```

Performance-wise I think it's the same and we should go with a cleaner approach which to me is the second one but i don't really have a strong opinion here

Reply:
I think performance wise the first one is worse. We should rely as much on orsjon as possible

Reply:
join is super slow